src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -142,11 +142,16 @@ protected function getAllowedAttributes($classOrObject, array $context, $attribu
             return false;
         }
 
-        if (null !== $this->classDiscriminatorResolver && null !== $discriminatorMapping = $this->classDiscriminatorResolver->getMappingForMappedObject($classOrObject)) {
-            $allowedAttributes[] = $attributesAsString ? $discriminatorMapping->getTypeProperty() : new AttributeMetadata($discriminatorMapping->getTypeProperty());
+        $class = is_object($classOrObject) ? \get_class($classOrObject) : $classOrObject;
```

Comment:
`is_object` should be fully-qualified here, to benefit from the optimized opcode.

Reply:
btw, shouldn't it go inside the `if (null !== $this->classDiscriminatorResolver) {` ?

Reply:
Good points, updated 👍